client/webserver/site/src/js/ws.ts

```diff
@@ -114,6 +112,16 @@ class MessageSocket {
         clearTimeout(timeout)
         conn = this.connection = null
         forward('close', null, this.handlers)
+
+        // Certain browsers have degrading performance bug when retries are issued
+        // perpetually (e.g. browser tab is still trying to reconnect WS to dexc after
+        // user shut it down but left his tab open). Refreshing page here is a patch
+        // for this behavior, it breaks this browser tab retry cycle.
+        if (retrys > 10) {
+          reloadPage()
+          return
+        }
```

Comment:
I don't have the exact data on this, but feel like limiting retries at 10 should significantly simplify it for many browsers, and it gives the user enough of time-buffer to auto-recover from temporary WS failures (beyond 10 attempts he'll have to refresh page manually).

Reply:
Mixed feeling, but I think this is fine. It is going to dump them to a browser error page if dexc is not running, where before, they would just be looking at an outdated view. Maybe that's better.